boards/makerbase/mks_canable_v20/doc/index.rst

```diff
@@ -0,0 +1,173 @@
+| GPIO      | on-chip    | gpio                                |
++-----------+------------+-------------------------------------+
+| PWM       | on-chip    | pwm                                 |
++-----------+------------+-------------------------------------+
+| FDCAN     | on-chip    | can                                 |
++-----------+------------+-------------------------------------+
+
+Other hardware features are not yet supported on this Zephyr port.
+
+The default configuration can be found in the defconfig file:
+:zephyr_file:`boards/makerbase/mks_canable_v20/mks_canable_v20_defconfig`
+
+
+Connections and IOs
+===================
+
+The stm32g431c8 mcu is the core of MKS CANable V2.0 board and it has 6 GPIO controllers.
+These controllers are responsible for pin muxing, input/output, pull-up, etc.
+
+For more details please refer to `MKS CANable V2.0 schematic`_.
+
+Default Zephyr Peripheral Mapping is fixed by `MKS CANable V2.0 schematic`_:
+----------------------------------------------------------------------------
+
+.. rst-class:: rst-columns
+
+- CAN_RX/BOOT0 : PB8
+- CAN_TX : PB9
+- D2 : PA15
+- D3 : PA0
+- USB_DN : PA11
+- USB_DP : PA12
+- SWDIO : PA13
+- SWCLK : PA14
+- NRST : PG10
+
+System Clock
+------------
+
+The MKS CANable V2.0 system clock is driven by internal high speed oscillator.
+By default system clock is driven by PLL clock at 170 MHz,
+the PLL is driven by the 16 MHz high speed internal oscillator.
+
+Programming and Debugging
+*************************
+
+MKS CANable V2.0 board includes an SWDIO debug connector header J4.
+
+.. note::
+
+   The debugger is not the part of the board!
+
+Applications for the ``mks_canable_v20`` board target can be built and
+flashed in the usual way (see :ref:`build_an_application` and
+:ref:`application_run` for more details).
+
+Flashing
+========
+
+The board is configured to be flashed using west OpenOCD runner.
+
+Flashing an application to MKS CANable V2.0
+-------------------------------------------
+
+The debugger shall be wired to MKS CANable V2.0 board's J4 connector
+according `MKS CANable V2.0 schematic`_.
+
+Build and flash an application. Here is an example for
+:zephyr:code-sample:`hello_world`.
+
+.. zephyr-app-commands::
+   :zephyr-app: samples/hello_world
+   :board: mks_canable_v20
+   :goals: build flash
+   :west-args: -S rtt-console
+   :compact:
+
+The argument ``-S rtt-console`` is needed for debug purposes with SEGGER RTT protocol.
+This option is optional and may be omitted. Omitting it frees up RAM space but prevents RTT usage.
+
+If option ``-S rtt-console`` is selected, the connection to the target can be established as follows:
+
+.. code-block:: console
+
+   $ telnet localhost 9090
+
+You should see the following message on the console:
+
+.. code-block:: console
+
+   $ Hello World! mks_canable_v20/stm32g431xx
+
+.. note::
+
+   Current OpenOCD config will skip Segger RTT for OpenOCD under 0.12.0.
```

Comment:
I would remove all of this and just refer to the common documentation on how to use Segger RTT.

Reply:
Somehow I can't find proper compact description for segger RTT usage same way it is done in my current documentation. Would you please put a link here?

Reply:
This one shows only KConfig entry and no step-by-step guide like i my documentation.

Reply:
Feel free to open a PR for improving the documentation on using Segger RTT console, but let's make it generic instead of board-specific (as there is really nothing board-specific to it).

Reply:
> Feel free to open a PR for improving the documentation on using Segger RTT console, but let's make it generic instead of board-specific (as there is really nothing board-specific to it).

Definitely I could take over with my current experience. Would you please suggest a documentation section (please a link to some index.rst file), where I can place the corresponding documentation?